libsolidity/formal/SMTEncoder.cpp

```diff
@@ -834,7 +834,21 @@ void SMTEncoder::endVisit(Literal const& _literal)
 	else if (smt::isBool(type))
 		defineExpr(_literal, smtutil::Expression(_literal.token() == Token::TrueLiteral ? true : false));
 	else if (smt::isStringLiteral(type))
+	{
 		createExpr(_literal);
+
+		// Add constraints for the length and values as it is known.
+		auto symbArray = dynamic_pointer_cast<smt::SymbolicArrayVariable>(m_context.expression(_literal));
+		solAssert(symbArray, "");
+
+		auto value = _literal.value();
+		m_context.addAssertion(symbArray->length() == smtutil::Expression(value.length()));
```

Comment:
```suggestion
		m_context.addAssertion(symbArray->length() == value.length());
```
Does that work?

Reply:
Yes, can simplify both assertions -- well need to check the tests if it does not mess it up :)

Reply:
Seems to work, pushed.